src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1539,31 +1539,22 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     // - caseSensitive: boolean that represents if the current search is case sensitive
     // Return Value:
     // - <none>
-    void ControlCore::Search(const winrt::hstring& text,
-                             const bool goForward,
-                             const bool caseSensitive)
+    void ControlCore::Search(const winrt::hstring& text, const bool goForward, const bool caseSensitive)
     {
-        if (text.size() == 0)
+        auto lock = _terminal->LockForWriting();
+        _terminal->SetBlockSelection(false);

```

Comment:
I moved it down to `TriggerSelection()`. I saw that other callers do it the same way. I was worried that `Search` might otherwise end up making block selections. But I tested it and it works.